libgraph/include/katana/GraphTopology.h

```diff
@@ -43,6 +45,59 @@ struct KATANA_EXPORT GraphTopologyTypes {
   using EdgeDestVec = NUMAArray<Node>;
   using PropIndexVec = NUMAArray<PropertyIndex>;
   using EntityTypeIDVec = NUMAArray<EntityTypeID>;
+  */
+
+  struct Node: public OpaqueIDLinear<Node, uint32_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct Edge: public OpaqueIDLinear<Edge, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct OutEdgeHandle: public OpaqueIDLinear<OutEdgeHandle, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct InEdgeHandle: public OpaqueIDLinear<InEdgeHandle, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct InOutEdgeHandle: public OpaqueIDLinear<InOutEdgeHandle, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  // TODO(Amber/emcginnis): Can this be uint32_t?
+  struct NodePropertyIndex: public OpaqueIDLinear<NodePropertyIndex, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  struct EdgePropertyIndex: public OpaqueIDLinear<EdgePropertyIndex, uint64_t> {
+    using OpaqueIDLinear::OpaqueIDLinear;
+  };
+
+  // TODO(Amber): Opaque IDs should define a separate iterator type instead of
+  // mixing the concepts in one class. 
+  using node_iterator = OpaqueIDCountingIterator<Node>;
+  using nodes_range = StandardRange<node_iterator>;
+
+  using edge_iterator = OpaqueIDCountingIterator<OutEdgeHandle>;
+  using edges_range = StandardRange<edge_iterator>;
+  // TODO(Amber): Remove because we have Nodes() method to iterate over all nodes
+  using iterator = node_iterator;
+
+  //TODO(emcginnis): Each of these *Vec types should really be *Array since they are not resizable
+  // NOTE(Amber): using underlying type so that (1) tusba & gluon partitioner interface doesn't
+  // need to change (2) We can repurpose the adj indices for InOutEdgeHandle in
+  // UndirectedTopology, and InEdgeHandle in Bidirectional topology 
+  using AdjIndexVec = NUMAArray<OutEdgeHandle::underlying_type>;
```

Comment:
A random thought as I'm reading the code - I'd alias `OutEdgeHandle::underlying_type` as smth like `OutEdgeHandle_t` and so on, too hard on the eyes, IMHO.

Reply:
Agree, I've had to do this in analytics code. However `_t` name is not specific enough. Let me suggest `OutEdgeHandleInt` and `NodeInt` and so on.